include/interfaces/hydrogels_one_field.h

```diff
@@ -0,0 +1,263 @@
+                              bool compute_only_system_terms) const;
+
+
+  /*void compute_system_operators(const DoFHandler<dim,spacedim> &,
+                                const std::vector<shared_ptr<LATrilinos::BlockMatrix> >,
+                                LinearOperator<LATrilinos::VectorType> &,
+                                LinearOperator<LATrilinos::VectorType> &) const;
+  */
+
+
+private:
+  double T;
+  double Omega;
+  double G;
+  double chi;
+  double l0;
+
+  double mu0;
+  double l02;
+  double l03;
+  double l0_3;
+  double l0_6;
+  const double R=8.314;
+
+
+  /*  mutable shared_ptr<TrilinosWrappers::PreconditionAMG> U_prec;
+    mutable shared_ptr<TrilinosWrappers::PreconditionJacobi> p_prec;
+    mutable shared_ptr<TrilinosWrappers::PreconditionJacobi> c_prec;
+  */
+
+};
+
+template <int dim, int spacedim, typename LAC>
+HydroGelOneField<dim,spacedim,LAC>::HydroGelOneField() :
+  PDESystemInterface<dim,spacedim,HydroGelOneField<dim,spacedim,LAC>, LAC>("Free Swelling One Field",
+      dim,1,
+      "FESystem[FE_Q(2)^d]",
+      "u,u,u","1")
+{}
+
+/*
+template <int dim, int spacedim, typename LAC>
+void
+HydroGelOneField<dim,spacedim,LAC>::
+set_matrix_couplings(std::vector<std::string> &couplings) const
+{
+  couplings[0] = "1,1,0;1,0,1;0,1,1";
+  couplings[1] = "0,0,0;0,1,0;0,0,0";
+}
+*/
+
+template <int dim, int spacedim, typename LAC>
+template <typename EnergyType, typename ResidualType>
+void
+HydroGelOneField<dim,spacedim,LAC>::
+energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                       FEValuesCache<dim,spacedim> &fe_cache,
+                       std::vector<EnergyType> &energies,
+                       std::vector<std::vector<ResidualType> > &residuals,
+                       bool compute_only_system_terms) const
+{
+  EnergyType alpha = this->alpha;
+
+  this->reinit(alpha, cell, fe_cache);
+
+  const FEValuesExtractors::Vector displacement(0);
+
+  auto &Fs = fe_cache.get_deformation_gradients("solution", "Fu", displacement, alpha);
+  auto &fev = fe_cache.get_current_fe_values();
+  auto &JxW = fe_cache.get_JxW_values();
+
+  const unsigned int n_q_points = JxW.size();
+
+  ResidualType rt = 0;
+  this->reinit (rt, cell, fe_cache);
+  auto &Fs_res = fe_cache.get_deformation_gradients("solution", "u", displacement, rt);
+
+  for (unsigned int q=0; q<n_q_points; ++q)
+    {
+      const Tensor<2, dim, EnergyType>  &F = Fs[q];
+      const Tensor<2, dim, EnergyType>   C = transpose(F)*F;
+
+      const EnergyType I = trace(C);
+      const EnergyType J = determinant(F);
+
+
+      EnergyType psi = ( 0.5*G*l0_3*(l02*I - dim)
+
+                         + (l0_3*R*T/Omega)*((J*l03-1.)*std::log((J*l03-1.)/(J*l03))
+                                             + chi*((J*l03-1.)/(J*l03)) )
+
+                         - (mu0)*(J*l03-1.)/(Omega*l03)) ;
+
+      energies[0] += psi*JxW[q];
+
+      auto &F_res = Fs_res[q];
```

Comment:
error: no member named 'val' in
      'dealii::Tensor<2, 3, Sacado::Fad::DFad<double> >'
      auto &F_res = Fs[q].val();